sklearn/preprocessing/data.py

```diff
@@ -656,7 +658,7 @@ def partial_fit(self, X, y=None):
             # First pass
             if not hasattr(self, 'n_samples_seen_'):
                 self.mean_ = .0
-                self.n_samples_seen_ = 0
+                self.n_samples_seen_ = np.zeros(X.shape[1], dtype=np.int32)
```

Comment:
Am I being overly cautious if I worry that this changes public API? We could squeeze this back to a single value if `np.ptp(n_samples_seen_) == 0` after each `partial_fit`

Reply:
I agree it's better to try to preserve backward compat but maybe instead of using a data dependent `np.ptp` we could do the squeeze only when `self.force_all_finite != 'allow-nan'`. This way the user is more in control and it's more explicit.

Reply:
In other feature-wise preprocessing, we're allowing NaNs through by default, under the assumption that extra processing cost is negligible and that downstream estimators will deal with or complain about the presence of NaN. Thus `self.force_all_finite` does not exist.

Reply:
Indeed, I misread the code snippet. We don't have much choice but to use the `np.ptp` trick then.